cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/upgrade/operation/RdsInstanceStatusesToRdsStateConverter.java

```diff
@@ -0,0 +1,44 @@
+package com.sequenceiq.cloudbreak.cloud.aws.connector.resource.upgrade.operation;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RdsInstanceStatusesToRdsStateConverter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RdsInstanceStatusesToRdsStateConverter.class);
+
+    public RdsState convert(Map<String, String> dbArnToInstanceStatusesMap) {
+        if (dbArnToInstanceStatusesMap.isEmpty()) {
+            LOGGER.debug("There is no info from RDS instances, returning {}", RdsState.UNKNOWN);
+            return RdsState.UNKNOWN;
+        }
+
+        Map<String, String> notAvailableInstances = getNotAvailableInstances(dbArnToInstanceStatusesMap);
+        if (notAvailableInstances.isEmpty()) {
+            LOGGER.debug("There are no unavailable RDS instances, returning {}", RdsState.AVAILABLE);
+            return RdsState.AVAILABLE;
+        }
+
+        if (notAvailableInstances.values().stream().anyMatch("upgrading"::equals)) {
```

Comment:
`upgrading` could be a string field in the enum

Reply:
In RdsState there are 2 states corresponding to actual RdsState, and 2 other that have only meaning in CB code. If I have enum consts like AVAILABLE("available"), then e.g. OTHER("other") would imply that "other" is also a valid RDS state whereas it is not. 

But I agree that this would need at least a constant, or maybe another enum, like AwsRdsState.

Reply:
That is fine for me, thanks